src/tools/tidy/src/lib.rs

```diff
@@ -208,10 +208,16 @@ pub fn ensure_version_or_cargo_install(
                     .chain(std::iter::once(tool_bin_dir.clone())),
             )
             .expect("build dir contains invalid char"),
-        )
-        .env("RUSTFLAGS", "-Copt-level=0")
-        .spawn()?
-        .wait()?;
+        );
+
+    // On CI, we set opt-level flag for quicker installation.
+    // But since lower opt-level decreases the tool's performance,
+    // We don't set this option on local.
+    if CiEnv::is_ci() {
+        cmd.env("RUSTFLAGS", "-Copt-level=1");
```

Comment:
Any reason for switching `opt-level=0` to `opt-level=1`? Not sure if that trade-off is worth it.

Reply:
CI has only four cores, so it would be great to benchmark it directly on CI. You can modify the CI config (like create temporary steps in `ci.yml`) to install the typos binary and then run it on the directories that typo spellchecks, to see the actual durations on CI.

Reply:
Thanks! I will try it then modify the line if necessary.

Reply:
I tested each opt-level in the linked PR. opt-level=0 is the fastest on the CI env.
```
0: 0m42.677s (installation) + 0m36.815s (execution) = 1m19s
1: 1m29.025s + 0m2.782s = 1m32s
2: 1m48.930s + 0m1.872s = 1m51s
3: 1m51.310s + 0m1.947s = 1m53s
```
Ref: https://github.com/rust-lang/rust/actions/runs/18260861226/job/51988569220?pr=147379



I will roll the opt-level back to zero.

Reply:
Thank you for performing the experiment. The ~10 secs difference isn't so large, but I think that 0 is still a reasonable default here.